sample/sample-with-di/app/src/main/java/com/dropbox/kaiken/sample/advancedsample/app/AdvancedKaikenSampleApplication.kt

```diff
@@ -1,33 +1,87 @@
+import com.dropbox.kaiken.skeleton.skeleton.dagger.SdkSpec
+import com.dropbox.kaiken.skeleton.skeleton.usermanagement.UserManager
+import com.dropbox.kaiken.scoping.AppScope
+import com.dropbox.kaiken.scoping.SingleIn
+import com.dropbox.kaiken.scoping.SkeletonScope
+import com.dropbox.kaiken.scoping.UserScope
+import com.dropbox.kaiken.scoping.cast
+import com.squareup.anvil.annotations.ContributesTo
+import com.squareup.anvil.annotations.MergeComponent
+import dagger.BindsInstance
+import dagger.Component
+import dagger.Module
+import dagger.Provides
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+import javax.inject.Inject
 
-class AdvancedKaikenSampleApplication : Application(), ScopedServicesProvider {
 
-    private lateinit var appServices: AdvancedKaikenSampleAppServices
+class AdvancedKaikenSampleApplication : SkeletonApplication() {
 
-    // Let's use a map to mimic our "UserManager".
-    private val userServicesMap = mutableMapOf<String, AdvancedKaikenSampleUserServices>()
+    override fun getSdkSpec(): SdkSpec = DaggerSkeletonComponent.factory().create(this) as DaggerSkeletonComponent
 
-    override fun onCreate() {
-        super.onCreate()
+    @Inject
+    lateinit var userFlow: @JvmSuppressWildcards MutableSharedFlow<String>
 
-        appServices = DaggerAdvancedKaikenSampleAppServices.factory().create()
+    @Inject
+    lateinit var userManager: UserManager
 
-        // Let's "log-in" in a user for sample purposes. In a real application this code would definitely not live
+    override fun onCreate() {
+        super.onCreate()
+        appServices.cast<ApplicationInjector>().inject(this)
+        // Let's "log-in" users for sample purposes. In a real application this code would definitely not live
         // here and would be way way more complex
-        val userProfile = UserProfile("awesome_user", "Awesome User")
+        GlobalScope.launch {
+            userFlow.collect {
+                userManager.setActiveUser(it)
+            }
+        }
 
-        userServicesMap["awesome_user"] = DaggerAdvancedKaikenSampleUserServices.factory().create(
-            appServices,
-            userProfile
-        )
+           GlobalScope.launch {
+               userFlow.emit("1")
+           }
+           GlobalScope.launch {
+               userManager.setActiveUser("1")
+
+           }
     }
+}
 
-    override fun provideAppServices(): AppServices = appServices
+@ContributesTo(AppScope::class)
+interface ApplicationInjector {
+    fun inject(application: AdvancedKaikenSampleApplication)
+}
 
-    override fun provideUserServicesOf(userId: String): UserServices? = userServicesMap.get(userId)
+@ContributesTo(UserScope::class)
+@Module
+class UserModule {
+    @Provides
+    @SingleIn(UserScope::class)
+    fun provideUserProfile(userId: Int) = UserProfile(userId.toString(), userId.toString())
```

Comment:
We should probably set this up to have an actual user name for clarity.

Reply:
making a todo for follow up diff, it would mean some larger changes as we only pass an ID through the fake interactor stream